src/cpp/session/modules/SessionFind.cpp

```diff
@@ -741,6 +719,38 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
       return error;
    }
 
+   void cleanLineAndGetMatches(std::string* pEncodedLine,
+                               json::Array* pMatchOn, json::Array* pMatchOff)
+   {
+      // The incoming string is assumed to have color encodings from the initial grep command.
+      // These encodings are parsed out and their positions are placed in pMatchOn and pMatchOff.
```

Comment:
Why do we put them into `pMatchOn` and `pMatchOff`? Are we removing the color encodings or just keeping track of them?

Reply:
We both remove them and keep track of them. These color encodings are placed by `grep` to indicate the matched word so we keep track of where they were to tell us the location of that matched word. We remove them since they're not part of the original line.

Reply:
What's the purpose of putting their locations into `pMatchOn` and `pMatchOff`? Is it for removing them later? Or is the `pMatchOn` right after the color open (i.e. the start of the actual match) and `pMatchOff` right before the color close?

Reply:
As we iterate over the line's matches we are storing `pMatchOn` and `pMatchOff` so we know where the matches are so we can later replace them. At the same time we are creating a string `cleanLine` that includes everything except for the color encodings. This string gets returned in the variable the original line with the encodings was passed in.